Sources/Private/CoreAnimation/Layers/TextLayer.swift

```diff
@@ -26,10 +26,11 @@ final class TextLayer: BaseCompositionLayer {
   override func setupAnimations(context: LayerAnimationContext) throws {
     try super.setupAnimations(context: context)
 
-    // Question 1: Which keypath to pass?
     _ = try context.currentKeypath.keys.map { key in
-      // Question 2: Where do we get the text layer text?
-      let customText = context.textProvider.textFor(keypathName: key, sourceText: "text layer text")
+      // Question 2.1: Should we get from somewhere source text? What if the user has custom name?
+      let customText = context.textProvider.textFor(keypathName: context.currentKeypath.fullPath, sourceText: "text layer text")
+      // Question 2.2: If we don't have this .isEmpty check, then how do we decide when to set our custom text, and when to
+      // set the result of the `exactlyOneKeyFrame()`?
```

Comment:
If you look at the behavior of the main thread rendering engine, it just unconditionally sets `renderLayer.text` to the result of `textProvider.textFor(keypathName:sourceText:)`. Since that method doesn't return an optional, it looks like expected usage is to just `return sourceText` if you don't want to customize the text for that specific keypath. That's how `DefaultTextProvider` works, for example.

Reply:
Updated the method in [c8a84c2](https://github.com/airbnb/lottie-ios/pull/1723/commits/c8a84c279c0cf68dadecf15363af36fa9e2029dd) 🟡 

Then we just set it like this? 🤔